core/store/indexstore/indexer.go

```diff
@@ -52,31 +38,40 @@ func CloseIndexer() {
 // after an entire section has been finished or in case of rollbacks that might
 // affect already finished sections.
 type Indexer struct {
-	backend bloomIndexer // Background processor generating the index data content
+	backend        bloomIndexer // Background processor generating the index data content
+	start          uint32       // Start is the start height of bloom filter supported
+	storedSections uint32       // Number of sections successfully indexed into the database
+}
 
-	update chan struct{} // Notification channel that headers should be processed
-	quit   chan struct{} // Quit channel to tear down running goroutines
+func New(dataDir string, curBlockHeight uint32) (*Indexer, error) {
+	db, err := leveldbstore.NewLevelDBStore(filepath.Join(dataDir, bloomIdxDir))
+	if err != nil {
+		return nil, err
+	}
 
-	storedSections uint32 // Number of sections successfully indexed into the database
-	processing     uint32 // Atomic flag whether indexer is processing or not
-}
+	backend := initBloomIndexer(db)
 
-func InitIndexer(dataDir string) error {
-	store, err := leveldbstore.NewLevelDBStore(filepath.Join(dataDir, bloomIdxDir))
+	start, err := backend.GetFilterStart()
 	if err != nil {
-		return err
+		if err != common3.ErrNotFound {
+			return nil, fmt.Errorf("new indexer: %s", err.Error())
+		}
+		err = backend.PutFilterStart(curBlockHeight)
```

Comment:
I do not think we need save all the zero bytes in db before the height evm feature enabled.